lib/elixir/lib/version.ex

```diff
@@ -389,8 +449,9 @@ defmodule Version do
   compiled match_spec, nor can it be stored on disk).
   """
   @spec compile_requirement(Requirement.t()) :: Requirement.t()
+  # TODO: Deprecate this function, it's a NOOP
```

Comment:
I would update the documentation to not talk about match specs.

> Compiles a requirement to its internal representation for faster matching.
>
> The internal representation is opaque.

In addition, should we include the Elixir version in which this function should be hard-deprecated?

Reply:
> I would update the documentation to not talk about match specs.

Right, I wasn't sure what to do here. The docs you suggest are also incorrect since it won't give faster matching. Since there is no reason to call the function anymore should we just soft-deprecate now?

> In addition, should we include the Elixir version in which this function should be hard-deprecated?

What is the usual policy for this? Soft-deprecate on second minor release and hard-deprecate on fourth? So remove docs on v1.12 and and add `@deprecated` on v1.14?

Reply:
I would keep this function. Change the docs to say "it may" optimize. The idea is, maybe today it can't do anything, but maybe in a year it can.

Reply:
> The idea is, maybe today it can't do anything, but maybe in a year it can.

Shouldn't we add the docs back in a year in that case? Or are you saying we shouldn't deprecate this function in the future?

Reply:
We shouldn't deprecate it at all IMO.